src/blockchain/pledges.js

```diff
@@ -268,6 +270,15 @@ const pledges = (app, liquidPledging) => {
       token,
     };
 
+    // Propagate comment
+    if (
+      donations.length === 1 &&
```

Comment:
@aminlatifi How can I test it on local environment?
I would like to be able to test all scenarios locally, 
anyway I think the code should work, I just can't figure out why we check `donations.length === 1` (because of my limited knowledge)

Reply:
You should had setup local version on your machine with @mahdine once, didn't you?

Reply:
here donations are parent donations of new donation which will be created. The case that inherits comment from parent is when user donates to a project, first a donation is created under the ownership of user (who has donated), then money is moved to another donation (under project ownership). In this case the second donation has only one parent.

Reply:
However length===1 is not necessary and can be deleted I guess

Reply:
@aminlatifi 

Thanks for your guide.

I tested in this way
1. First I donated to my campaign and  donationatio created with my comment and `ownerType:giver`

```
{
    "_id" : ObjectId("5fb9351e1e6de9dbe8cbc56b"),
    "status" : "Waiting",
    "mined" : true,
    "parentDonations" : [],
    "isReturn" : false,
    "lessThanCutoff" : true,
    "usdValue" : 0,
    "giverAddress" : "0xFFcf8FDEE72ac11b5c542428B35EEF5769C409f0",
    "amount" : "2000000000000000000",
    "amountRemaining" : "0",
    "pledgeId" : "7",
    "homeTxHash" : "0x6f79abd0e2b06ca48d46a32d7bdcbff99e89d477100e73e3fd850bb71aa0a736",
    "txNonce" : 7,
    "token" : {
        "_id" : ObjectId("5fb935471e6de9dbe8cbc570"),
        "name" : "MiniMe Test Token",
        "address" : "0xe78A0F7E598Cc8b0Bb87894B0F60dD2a88d6a8Ab",
        "foreignAddress" : "0x8F086f895deBc23473dfe507dd4BF35D6184552c",
        "symbol" : "ANT",
        "decimals" : "3"
    },
    "comment" : "Testing propagation comments",
    "ownerType" : "giver",
    "ownerTypeId" : "0xFFcf8FDEE72ac11b5c542428B35EEF5769C409f0",
    "ownerId" : NumberLong(2),
    "campaignId" : "5fb9205ff2afca9841c2d5e8",
    "createdAt" : ISODate("2020-11-21T15:41:18.412Z"),
    "updatedAt" : ISODate("2020-11-21T15:42:03.499Z"),
    "__v" : 0,
    "actionTakerAddress" : "0x610Bb1573d1046FCb8A70Bbbd395754cD57C2b60",
    "intendedProjectId" : NumberLong(0),
    "txHash" : "0x13d57a1f854ee822dd04003c58a6d455480bdf1e8ff646fee1b4109f54d15500"
}
```
2. Then a donation created with `ownerType:campaign` that has the parent's comment

```
{
    "_id" : ObjectId("5fb935521e6de9dbe8cbc571"),
    "status" : "Committed",
    "mined" : true,
    "parentDonations" : [ 
        "5fb9351e1e6de9dbe8cbc56b"
    ],
    "isReturn" : false,
    "lessThanCutoff" : false,
    "usdValue" : 0,
    "amount" : "2000000000000000000",
    "amountRemaining" : "2000000000000000000",
    "giverAddress" : "0xFFcf8FDEE72ac11b5c542428B35EEF5769C409f0",
    "ownerId" : NumberLong(5),
    "ownerTypeId" : "5fb9205ff2afca9841c2d5e8",
    "ownerType" : "campaign",
    "pledgeId" : "8",
    "commitTime" : ISODate("2020-11-21T15:41:52.000Z"),
    "createdAt" : ISODate("2020-11-21T15:41:52.000Z"),
    "txHash" : "0x13d57a1f854ee822dd04003c58a6d455480bdf1e8ff646fee1b4109f54d15500",
    "token" : {
        "_id" : ObjectId("5fb935521e6de9dbe8cbc572"),
        "name" : "MiniMe Test Token",
        "address" : "0xe78A0F7E598Cc8b0Bb87894B0F60dD2a88d6a8Ab",
        "foreignAddress" : "0x8F086f895deBc23473dfe507dd4BF35D6184552c",
        "symbol" : "ANT",
        "decimals" : "3"
    },
    "comment" : "Testing propagation comments",
    "homeTxHash" : "0x6f79abd0e2b06ca48d46a32d7bdcbff99e89d477100e73e3fd850bb71aa0a736",
    "actionTakerAddress" : "0x610Bb1573d1046FCb8A70Bbbd395754cD57C2b60",
    "updatedAt" : ISODate("2020-11-21T15:41:52.000Z"),
    "__v" : 0
}
```

3. At the end when I delegated money from campaign to milestone created a donation with `ownerType:milestone` that doesn't have the any comment, is that correct?( If its needed to comment be here also your code doesnt work well but if comments just needed to propagate to first child not anymore it works fine:

```
{
    "_id" : ObjectId("5fb935f795738bdcf43672d1"),
    "status" : "Committed",
    "mined" : true,
    "parentDonations" : [ 
        "5fb92635262264d8b407c036"
    ],
    "isReturn" : false,
    "lessThanCutoff" : false,
    "usdValue" : 0,
    "txHash" : "0x223e370e331b18314ea5a7442cefea338e9f94d0a9e3d7d7162d9f2fe363e519",
    "amount" : "130000000000000",
    "amountRemaining" : "130000000000000",
    "giverAddress" : "0xFFcf8FDEE72ac11b5c542428B35EEF5769C409f0",
    "pledgeId" : "5",
    "token" : {
        "_id" : ObjectId("5fb9360395738bdcf43672d6"),
        "name" : "Home Ganache ETH",
        "address" : "0x0",
        "foreignAddress" : "0x5a42ca500aB159c51312B764bb25C135026e7a31",
        "symbol" : "ETH",
        "decimals" : "6"
    },
    "ownerId" : NumberLong(8),
    "ownerTypeId" : "5fb92677262264d8b407c039",
    "ownerType" : "milestone",
    "actionTakerAddress" : "0xFFcf8FDEE72ac11b5c542428B35EEF5769C409f0",
    "createdAt" : ISODate("2020-11-21T15:44:55.923Z"),
    "updatedAt" : ISODate("2020-11-21T15:45:07.283Z"),
    "__v" : 0,
    "commitTime" : ISODate("2020-11-21T15:44:57.000Z")
}
```

Reply:
Thanks for testing, yes it's correct